### PR TITLE
feat/back-end/chat

### PR DESCRIPTION
**Pull request by park20011029**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our convention
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Feature
- [ ] Bugfix
- [ ] Refactor (no functional changes, no api changes)
- [ ] Design changes
- [ ] Comment added
- [ ] Code style update (formatting, local variables)
- [ ] Test code added
- [ ] Chore (other changes)
- [ ] Init
- [ ] Rename
- [ ] Remove
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

닉네임 중복 검사
채팅 기능 추가

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: server/src/main/java/project/manager/server/controller/UserController.java
Comment (SangminHann) [inline server/src/main/java/project/manager/server/controller/UserController.java:1]: commit message 작성에 신경써주세요

File: server/src/main/java/project/manager/server/service/UserService.java
Comment (SangminHann) [inline server/src/main/java/project/manager/server/service/UserService.java:1]: commit message 작성에 신경써주세요 *2

**Review by SangminHann — CHANGES_REQUESTED** [review]
커밋할 때 feat랑 chore 구분해서 하면 좋을 듯

**Review by SangminHann — APPROVED** [review]

**Review by JunSeop00 — APPROVED** [review]
LGTM

**Review by Kimdoodle — APPROVED** [review]
LGTM

**Pull request merged**